iocore/net/Connection.cc

```diff
@@ -45,7 +45,7 @@ get_listen_backlog()
   int listen_backlog;
 
   REC_ReadConfigInteger(listen_backlog, "proxy.config.net.listen_backlog");
-  return listen_backlog >= 0 ? listen_backlog : ats_tcp_somaxconn();
+  return (0 < listen_backlog && listen_backlog <= 65535) ? listen_backlog : ats_tcp_somaxconn();
```

Comment:
I found this but I'm not sure whether it really works. It wouldn't work as expected if `[0-65535]` was a regex. But if it worked, we should use it.
https://github.com/apache/trafficserver/blob/0456a54b8a8e9ba8dec326db2e410d30bf96b371/mgmt/RecordsConfig.cc#L747

Reply:
Currently , the range is -1 to 65535. But `[-1-65535]` is not accepted.
And the valid range is -1 and [1-65535], 0 is a bad value.

With this PR, if proxy.config.net.listen_backlog is set to 0, the result is the same as -1.

Reply:
Looks reasonable.  Poor selection of default values from way back when.